solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -367,11 +356,11 @@ library DepositUtils {
     /// @notice     Deletes state after termination of redemption process.
     /// @dev        We keep around the redeemer address so we can pay them out.
     function redemptionTeardown(Deposit storage _d) public {
-        // don't 0 redeemerAddress because we use it to calculate auctionTBTCAmount
         _d.redeemerOutputScript = "";
         _d.initialRedemptionFee = 0;
         _d.withdrawalRequestTime = 0;
         _d.lastRequestedDigest = bytes32(0);
+        _d.redeemerAddress = address(0);
```

Comment:
This affects my sinister plan above. I kind of like keeping the redeemer address as part of the final state of a deposit though… It means a closed deposit knows its original lot size, its final owner, and its redeemer. Feels clean somehow? Though I realize this may make the final tx cheaper 🤔

Reply:
It also feels like cleaning this up could inadvertently break some stuff that isn't worth spending time on fixing…

Reply:
I'd rather save the gas than have a redeemer value in storage. 

> It also feels like cleaning this up could inadvertently break some stuff that isn't worth spending time on fixing…

when we reclaim this state we are always wrapping up. as long as the state is reclaimed after its last use there should be no problem

Reply:
We're also assuming the gas saved will be saved, since there's a limit to how much gets refunded via state release. The transactions that do this aren't super expensive.

Reply:
that's a fair point, but we're definitely not reaching the cap for functions like `provideRedemptionProof`. We'll surely see the refund there